src/library.js

```diff
@@ -1328,6 +1351,52 @@ const createPilingJs = (rootElement, initOptions = {}) => {
       stateUpdates.add('layout');
     }
 
+    if (state.itemOpacity !== newState.itemOpacity) {
+      if (newState.itemOpacity) {
+        pileInstances.forEach(pile => {
+          pile.itemContainer.children.forEach((item, i) => {
+            item.alpha = isFunction(newState.itemOpacity)
+              ? newState.itemOpacity(item, pile, i)
+              : newState.itemOpacity;
+          });
+        });
+      }
+    }
+
+    if (state.pileOpacity !== newState.pileOpacity) {
+      if (newState.pileOpacity) {
+        pileInstances.forEach(pile => {
```

Comment:
As above, let's create a dedicated function called `updatePileStyle`, which handles all the pile styling (i.e., `pileOpacity`, `pileScale `, and `pileBorderSize ` for now). Additionally, when the piles state changed, after we called `updatePileItem()` and `updatePileLocation()`, we should then also call `updatePileStyle`.